api/Silk.Api.Domain/Feature/Infractions/AddInfraction.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentValidation;
+using MediatR;
+using Silk.Api.Data;
+using Silk.Api.Data.Models;
+
+namespace Silk.Api.Domain.Feature.Infractions
+{
+	public static class AddInfraction
+	{
+		public sealed record Request : IRequest<ApiModel>
+		{
+			public ulong TargetUserId { get; init; }
+			public ulong EnforcerUserId { get; init; }
+			public ulong GuildCreationId { get; init; }
+			
+			public InfractionType Type { get; init; }
+			
+			public DateTime Created { get; init; }
+			public DateTime? Expires { get; init; }
+		
+			public string Reason { get; init; }
+			
+			[JsonIgnore]
+			public string AddedBy { get; init; }
+		}
+
+		public sealed record ApiModel
+		{
+			public Guid  Key { get; init; }
+
+			public InfractionType Type { get; init; }
+			public ulong TargetUserId { get; init; }
+			public ulong EnforcerUserId { get; init; }
+			public ulong GuildCreationId { get; init; }
+		
+			public DateTime Created { get; init; }
+			public DateTime? Expires { get; init; }
+		
+			public string Reason { get; init; }
+		}
+		
+		public sealed class Handler : IRequestHandler<Request, ApiModel>
+		{
+			private readonly ApiContext _db;
+			public Handler(ApiContext db) => _db = db;
+
+			public async Task<ApiModel> Handle(Request request, CancellationToken cancellationToken)
+			{
+				var entity = new InfractionEntity()
+				{
+					Key = Guid.NewGuid(),
+					Type = request.Type,
+					TargetUserId = request.TargetUserId,
+					EnforcerUserId = request.EnforcerUserId,
+					GuildCreationId = request.GuildCreationId,
+					Created = request.Created,
+					Expires = request.Expires,
+					Reason = request.Reason,
+					AddedByFK = request.AddedBy
+				};
+
+				_db.Add(entity);
+				await _db.SaveChangesAsync(cancellationToken);
+
+				return ToModel(entity);
+			}
+
+			private static ApiModel ToModel(InfractionEntity entity)
+			{
+				return new()
+				{
+					Key = entity.Key,
+					Type = entity.Type,
+					TargetUserId = entity.TargetUserId,
+					EnforcerUserId = entity.EnforcerUserId,
+					GuildCreationId = entity.GuildCreationId,
+					Created = entity.Created,
+					Expires = entity.Expires,
+					Reason = entity.Reason
+				};
+			}
+		}
+
+		public sealed class Validator : AbstractValidator<Request>
+		{
+			public Validator()
+			{
+				RuleFor(r => r.TargetUserId)
+					.NotEmpty()
+					.WithMessage("targetUserId: must not be empty.");
```

Comment:
Can use `{PropertyName}` in the message to have it use the name of the property

Reply:
Yes but then it's PascalCase, which isn't great